### PR TITLE
017 pagination

### PR DESCRIPTION
**Pull request by Adnanhoq**

### CODE REVIEW

**Review by sdagn1 — DISMISSED** [review]

File: src/test/java/org/example/controller/JobRolesControllerTest.java [hunk 1]
```diff
@@ -47,27 +48,27 @@ void GetJobRoles_shouldReturnSuccessfulResponse_whenOpenJobRolesReturned() throw
...
     void getJobRoles_shouldReturnServerError_whenServiceFailsToGetJobRoles() throws
-            SQLException, DoesNotExistException {
-        Mockito.when(jobRoleService.getAllJobRoles()).thenThrow(SQLException.class);
-        Response response = jobRoleController.getJobRoles();
+            SQLException, DoesNotExistException, InvalidPageLimitException {
+        Mockito.when(jobRoleService.getAllJobRoles(1,10)).thenThrow(SQLException.class);
+        Response response = jobRoleController.getJobRoles(1,10);
 
         assertEquals(500, response.getStatus());
     }
```
Comment (JennicaMistry) [inline src/test/java/org/example/controller/JobRolesControllerTest.java:66]: You have not tested the case where the InvalidPageLimitException is thrown. What scenario would this exception be thrown?

Comment (JennicaMistry) [inline src/test/java/org/example/controller/JobRolesControllerTest.java:66]: You have not tested the case where the InvalidPageLimitException is thrown. What scenario would this exception be thrown?

**Review by JennicaMistry — CHANGES_REQUESTED** [review]

File: src/test/java/org/example/controller/JobRolesControllerTest.java (hunk 1 above)
Comment (Adnanhoq) [reply src/test/java/org/example/controller/JobRolesControllerTest.java:66]: Added

**Review by JennicaMistry — DISMISSED** [review]

File: src/test/java/org/example/integration/JobRolesIntegrationTest.java [hunk 2]
```diff
@@ -79,4 +79,116 @@ void getJobRoleById_shouldReturnAJobRole() {
...
+
+        );
+        String token = "Bearer " + client
+                .target("http://localhost:8080/api/auth/login")
+                .request()
+                .post(Entity.json(loginRequest))
+                .readEntity(String.class);
+
+        Response response = client
+                .target("http://localhost:8080/api/job-roles?page=1&limit=10")
```
Comment (sdagn1) [inline src/test/java/org/example/integration/JobRolesIntegrationTest.java]: Using a hardcoded variable on the address will break the integration tests as live uses https and local uses http.

Comment (Adnanhoq) [reply src/test/java/org/example/integration/JobRolesIntegrationTest.java]: Fixed

**Review by sdagn1 — APPROVED** [review]

**Review by JennicaMistry — APPROVED** [review]

**Pull request merged**